src/Symfony/Bundle/FrameworkBundle/ControllerMetadata/CachedArgumentMetadataFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\ControllerMetadata;
+
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadataFactoryInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ControllerMetadataFactoryInterface;
+
+/**
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class CachedArgumentMetadataFactory implements ArgumentMetadataFactoryInterface
```

Comment:
When I created it, the cache-warmer had only 3 lines of code. I'm not 100% sure of the setup yet (hence [WiP]), if I were to put everything in this class, the dependency on the ControllerResolver will cause a circular dependency.